api/src/main/java/org/openmrs/module/attachments/obs/ImageAttachmentHandler.java

```diff
@@ -65,21 +71,30 @@ protected boolean deleteComplexData(Obs obs, AttachmentComplexData complexData)
 	
 	@Override
 	protected ValueComplex saveComplexData(Obs obs, AttachmentComplexData complexData) {
+		int imageHeight = Integer.MAX_VALUE;
+		int imageWidth = Integer.MAX_VALUE;
+		String savedFileName = null;
 		
 		// We invoke the parent to inherit from the file saving routines.
 		obs = getParent().saveObs(obs);
 		
 		File savedFile = AbstractHandler.getComplexDataFile(obs);
-		String savedFileName = savedFile.getName();
```

Comment:
Restore that line and get rid of line 76, avoid `null` initializations everywhere possible.